src/components/Select/Select.tsx

```diff
@@ -0,0 +1,357 @@
+import { Listbox } from '@headlessui/react';
+import clsx from 'clsx';
+import type { ReactNode, ElementType } from 'react';
+import React, { useContext } from 'react';
+import styles from './Select.module.css';
+
+import type { ExtractProps } from '../../util/utility-types';
+
+import Icon from '../Icon';
+
+import PopoverContainer from '../PopoverContainer';
+import PopoverListItem from '../PopoverListItem';
+
+export type OptionsAlignType = 'left' | 'right';
+export type VariantType = 'compact' | 'full';
+
+type RenderProp<Arg> = (arg: Arg) => ReactNode;
+type PropsWithRenderProp<RenderPropArg> = {
+  children?: ReactNode | RenderProp<RenderPropArg>;
+  className?: string;
+  as?: ElementType;
+};
+
+type SelectProps = ExtractProps<typeof Listbox> & {
+  /**
+   * Screen-reader text for the select's label.
+   *
+   * When possible, use a visible label through the `labelText` prop or
+   * by passing a <Select.Label> into `chidren`. In rare cases where there's no
+   * visible label, you must provide an `aria-label` for screen readers.
+   * If you pass in an `aria-label`, you don't need `labelText` or <Select.Label>.
+   */
+  'aria-label'?: string;
+  /**
+   * Optional className for additional styling.
+   */
+  className?: string;
+  /**
+   * The style of the select.
+   *
+   * Compact renders select trigger button that is only as wide as the content.
+   */
+  variant?: VariantType;
+  /**
+   * Align select's popover to the left (default) or right of the trigger button.
+   */
+  optionsAlign?: OptionsAlignType;
+  /**
+   * Optional className for additional options menu styling.
+   *
+   * When not using the compact variant, if optionsClassName is provided please
+   * include the width property to define the options menu width.
+   */
+  optionsClassName?: string;
+};
+
+function childrenHaveLabelComponent(children?: ReactNode): boolean {
```

Comment:
Would this work with React fragments?

Reply:
Some of the factoring in this came from the existing tests I inherited here. I was trying to change only a little of that code, but these functions obfuscate what is being tested a little bit, in general.

I suspect there is something in react that actually does this (or a better job of it), so was inclined to leave it for a later investigation